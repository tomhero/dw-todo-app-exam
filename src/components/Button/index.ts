export { default as Button } from './CommonButton/CommonButton';
export { default as ActionButton } from './ActionButton/ActionButton';

export type { ActionItem } from './ActionButton/ActionButton';
