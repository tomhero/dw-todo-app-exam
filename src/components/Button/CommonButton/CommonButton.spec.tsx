import { fireEvent, render } from '@test';
import Button from './CommonButton';

describe('Component testing with testing-library', () => {
  it('should render button tag with given text', () => {
    const mockingText = 'this is button noting else.';
    const { container, getByText } = render(<Button>{mockingText}</Button>);
    expect(container).toMatchSnapshot();
    expect(getByText(mockingText)).toBeDefined();
  });

  it('should render any element as children properly', () => {
    const mockingComponent = (
      <div data-testid="child-component">
        <i>any</i> component
      </div>
    );
    const { container, getByTestId } = render(<Button>{mockingComponent}</Button>);
    expect(container).toMatchSnapshot();
    expect(getByTestId('child-component')).toBeDefined();
  });

  it('should render with correct given class name', () => {
    // Arrange
    const mockingClassName = 'zap-zap-class';

    // Act
    const { getByRole } = render(<Button className={mockingClassName}>test class button</Button>);

    // Assert
    expect(getByRole('button').classList.contains(mockingClassName)).toBeTruthy();
  });

  it('should be clickable', () => {
    const mockFn = jest.fn();
    const { getByRole } = render(<Button onClick={mockFn} />);

    const btn = getByRole('button');
    fireEvent.click(btn);

    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
