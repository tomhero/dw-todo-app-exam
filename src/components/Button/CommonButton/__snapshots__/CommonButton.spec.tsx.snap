// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing with testing-library should render any element as children properly 1`] = `
<div>
  <button
    class="dw-button "
  >
    <div
      data-testid="child-component"
    >
      <i>
        any
      </i>
       component
    </div>
  </button>
</div>
`;

exports[`Component testing with testing-library should render button tag with given text 1`] = `
<div>
  <button
    class="dw-button "
  >
    this is button noting else.
  </button>
</div>
`;
